test/statetest/statetest.hpp

```diff
@@ -59,8 +63,52 @@ struct StateTransitionTest
     std::unordered_map<uint64_t, std::string> input_labels;
 };
 
+template <typename T>
+T from_json(const json::json& j) = delete;
+
+template <>
+state::BlockInfo from_json<state::BlockInfo>(const json::json& j);
+
+template <>
+state::State from_json<state::State>(const json::json& j);
+
+template <>
+state::Transaction from_json<state::Transaction>(const json::json& j);
+
 StateTransitionTest load_state_test(const fs::path& test_file);
 
 void run_state_test(const StateTransitionTest& test, evmc::VM& vm);
 
+namespace utils
+{
+
+inline std::string shortest_hex(uint64_t v, bool add_prefix = true)
+{
+    if (v == 0)
+    {
+        return add_prefix ? "0x0" : "0";
+    }
+
+    auto value = intx::hex(intx::uint256{v});
+
+    // String leading zeroes.
+    if (size_t pos = value.find_first_not_of("0"); pos != std::string::npos)
+    {
+        value = value.substr(pos);
+    }
+    return (add_prefix ? "0x" : "") + value;
+}
+
+inline std::string hex(bytes_view bs, bool add_prefix = true)
+{
+    return (add_prefix ? "0x" : "") + evmc::hex(bs);
+}
+
+inline std::string hex(const intx::uint<256>& v, bool add_prefix = true)
```

Comment:
It looks we are just wrapping existing `hex` functions to add a prefix. How about a template:
```cpp
template<typename T>
auto hex0x(const T& v)
{
    return "0x" + hex(v);
}
```

Reply:
done